utilities/transactions/optimistic_transaction.cc

```diff
@@ -54,14 +55,33 @@ Status OptimisticTransaction::Prepare() {
 }
 
 Status OptimisticTransaction::Commit() {
-  // Set up callback which will call CheckTransactionForConflicts() to
-  // check whether this transaction is safe to be committed.
-  OptimisticTransactionCallback callback(this);
+  auto txn_db_impl =
+    static_cast_with_check<OptimisticTransactionDBImpl,  OptimisticTransactionDB>(txn_db_);
+  assert(txn_db_impl);
+  const size_t space = txn_db_impl->GetLockBucketsSize();
+
+  // NOTE(deyukong): we guarantee that we take the bucket lock in ascending
+  // order, so that we will never deadlock.
+  std::set<size_t> lk_idxes;
+  std::vector<std::unique_lock<std::mutex>> lks;
+  static murmur_hash hash;
```

Comment:
Use RocksDB's standard NPHash64(). Or, if you want, make buckets 2^n and use shifting. But it may not be worth it.